src/VisualStudio/Core/Def/Packaging/PackageInstallerServiceFactory.cs

```diff
@@ -482,6 +512,8 @@ private void OnWorkspaceChanged(object sender, WorkspaceChangeEventArgs e)
         private async Task<T?> PerformNuGetProjectServiceWorkAsync<T>(
             Func<INuGetProjectService, CancellationToken, ValueTask<T?>> doWorkAsync, CancellationToken cancellationToken)
         {
+            await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
             var serviceContainer = (IBrokeredServiceContainer?)await _asyncServiceProvider.GetServiceAsync(typeof(SVsBrokeredServiceContainer)).ConfigureAwait(false);
```

Comment:
But because of that caveat, we recommend folks switch to using [the extension method](https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.shell.serviceextensions.getserviceasync?view=visualstudiosdk-2019#Microsoft_VisualStudio_Shell_ServiceExtensions_GetServiceAsync__2_Microsoft_VisualStudio_Shell_IAsyncServiceProvider_System_Boolean_):
```suggestion
            IBrokeredServiceContainer? serviceContainer = await _asyncServiceProvider.GetServiceAsync<SVsBrokeredServiceContainer, IBrokeredServiceContainer>(throwOnFailure: false).ConfigureAwait(false);
```

Reply:
> No. GetServiceAsync can be called from a background thread. But it's generally not safe to cast the result as you're doing unless you are on the main thread or know that the service object is an MTA object. In this case, it is an MTA so you're fine.

Ah ok.

Reply:
> In this case, it is an MTA so you're fine.

@AArnott I wasn't able to find the documentation for this. Can you link it?